.github/workflows/deploy.yml

```diff
@@ -7,44 +7,38 @@ on:
         - staging
 
 jobs:
-    deploy_instance_1:
+    deploy_instance_staging:
         if: github.ref == 'refs/heads/staging'
         runs-on: ubuntu-latest
 
         steps:
             - name: Checkout code
               uses: actions/checkout@v3
+            
+            - name: Setup SSH
+              run: |
+                mkdir -p ~/.ssh
+                echo "${{ secrets.SERVER_KEY }}" > ~/.ssh/id_ed25519 && chmod 600 ~/.ssh/id_ed25519
```

Comment:
Why don't we revert to using a previous ssh-action version that works if v1.0.3 was causing issues? Maybe this has a security risk as for one it copies the secret key to the droplet making it potentially accessible, and it might also overwrite any secret key available on that file, making the droplet inaccessible? 

cc @ulope

Reply:
No that's not what's happening here. 
This code above is running in the github action worker. So the key never touches the server.
Since the worker container is destroyed after each run there should be no security issue.

On the other hand I'd say that using a random unaudited github action from a third party [that downloads unknown binaries](https://github.com/appleboy/ssh-action/blob/bb32884d53eb81279b477ae19ecf3d319978c768/entrypoint.sh#L62-L64) is a much greater security risk.